src/addons/keyboard_host.cpp

```diff
@@ -0,0 +1,197 @@
+#include "addons/keyboard_host.h"
+#include "storagemanager.h"
+
+#include "pio_usb.h"
+
+GamepadState _keyboard_host_state;
+
+KeyboardButtonMapping *_keyboard_host_mapDpadUp;
+KeyboardButtonMapping *_keyboard_host_mapDpadDown;
+KeyboardButtonMapping *_keyboard_host_mapDpadLeft;
+KeyboardButtonMapping *_keyboard_host_mapDpadRight;
+KeyboardButtonMapping *_keyboard_host_mapButtonB1;
+KeyboardButtonMapping *_keyboard_host_mapButtonB2;
+KeyboardButtonMapping *_keyboard_host_mapButtonB3;
+KeyboardButtonMapping *_keyboard_host_mapButtonB4;
+KeyboardButtonMapping *_keyboard_host_mapButtonL1;
+KeyboardButtonMapping *_keyboard_host_mapButtonR1;
+KeyboardButtonMapping *_keyboard_host_mapButtonL2;
+KeyboardButtonMapping *_keyboard_host_mapButtonR2;
+KeyboardButtonMapping *_keyboard_host_mapButtonS1;
+KeyboardButtonMapping *_keyboard_host_mapButtonS2;
+KeyboardButtonMapping *_keyboard_host_mapButtonL3;
+KeyboardButtonMapping *_keyboard_host_mapButtonR3;
+KeyboardButtonMapping *_keyboard_host_mapButtonA1;
+KeyboardButtonMapping *_keyboard_host_mapButtonA2;
+
+bool KeyboardHostAddon::available() {
+	return Storage::getInstance().getAddonOptions().keyboardHostOptions.enabled;
+}
+
+void KeyboardHostAddon::setup() {
+  KeyboardMapping& keyboardMapping = Storage::getInstance().getAddonOptions().keyboardHostOptions.mapping;
```

Comment:
This can be a `const` reference. You are not modifying the `struct` behind the reference.